ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java

```diff
@@ -1234,4 +1236,46 @@ public void shouldParseMultiLineWithInlineBracketedComments() {
     assertThat(statements, hasSize(1));
     assertThat(statements.get(0).getStatement(), is(instanceOf(ListStreams.class)));
   }
+
+  @Test
+  public void shouldBuildSimpleCaseStatement() {
+    // Given:
+    final String statementString =
+        "CREATE STREAM S AS SELECT CASE WHEN orderunits < 10 THEN 'small' WHEN orderunits < 100 THEN 'medium' ELSE 'large' END FROM orders;";
+
+    // When:
+    final Statement statement = KSQL_PARSER.buildAst(statementString, metaStore).get(0)
+        .getStatement();
+
+    // Then:
+    final Query query = ((CreateStreamAsSelect) statement).getQuery();
```

Comment:
Move these four lines into a helper method, e.g., `getSearchedCaseExpression()`, or maybe `getSearchedCaseExpressionFromCsas()`, since they are duplicated again in the test below, and currently make the tests a little harder to read.

Reply:
They are not exactly the same. One has `ELSE` and and the other one does not.

Reply:
I think I wasn't clear about my suggestion -- I agree the two tests are different and both add value. My suggestion was to make the following helper method:
```
private static SearchedCaseExpression getSearchedCaseExpressionFromCsas(final Statement statement) {
    final Query query = ((CreateStreamAsSelect) statement).getQuery();
    final QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
    final Expression caseExpression = ((SingleColumn) querySpecification.getSelect().getSelectItems().get(0)).getExpression();
    return (SearchedCaseExpression) caseExpression;
}
```
and use this helper method in each of the two tests you added, to make the tests easier to read (and reduce duplicated code). What do you think?

Reply:
Aha, I thought you refer to the query text. Agreed, it is more concise to have those in a method. Moved them to a method as you suggested.